ui/test/nuxeo-image-viewer.test.js

```diff
@@ -0,0 +1,326 @@
+        expect(fitToViewerOption.getAttribute('data-action')).to.equal('fit-to-viewer');
+
+        sinon.spy(viewer._el, 'zoomTo');
+        tap(fitToViewerOption);
+        expect(viewer._el.zoomTo.calledOnce).to.be.true;
+      });
+
+      test('Should call method to zoom in when third button is used', () => {
+        const zoomInOption = toolbar.children[2];
+        expect(zoomInOption.getAttribute('data-action')).to.equal('zoom-in');
+
+        sinon.spy(viewer._el, 'zoom');
+        tap(zoomInOption);
+        expect(viewer._el.zoom.withArgs(0.1).calledOnce).to.be.true;
+      });
+
+      test('Should call method to rotate left when fourth button is used', () => {
+        const rotateLeftOption = toolbar.children[3];
+        expect(rotateLeftOption.getAttribute('data-action')).to.equal('rotate-left');
+
+        sinon.spy(viewer._el, 'rotate');
+        tap(rotateLeftOption);
+        expect(viewer._el.rotate.withArgs(-90).calledOnce).to.be.true;
+      });
+
+      test('Should call method to rotate right when fifth button is used', () => {
+        const rotateRightOption = toolbar.children[4];
+        expect(rotateRightOption.getAttribute('data-action')).to.equal('rotate-right');
+
+        sinon.spy(viewer._el, 'rotate');
+        tap(rotateRightOption);
+        expect(viewer._el.rotate.withArgs(90).calledOnce).to.be.true;
+      });
+    });
+
+    suite('Window Resize', () => {
+      test("Should take no action when viewer's parent width is zero", async () => {
+        const container = await fixture(
+          html`
+            <div id="parent" style="width: 0px">
+              <nuxeo-image-viewer src="${base}/resources/sample.png" />
+            </div>
+          `,
+        );
+        const viewer = container.querySelector('nuxeo-image-viewer');
+        await viewerLoad(viewer);
+
+        sinon.spy(viewer._el, 'resize');
+        sinon.spy(viewer._el, 'zoomTo');
+
+        viewer.dispatchEvent(new CustomEvent('resize', { bubbles: true }));
+
+        expect(viewer._el.resize.notCalled).to.be.true;
+        expect(viewer._el.zoomTo.notCalled).to.be.true;
+      });
+
+      test("Should take no action when viewer's parent height is zero", async () => {
+        const container = await fixture(
+          html`
+            <div id="parent" style="height: 0px">
+              <nuxeo-image-viewer src="${base}/resources/sample.png" />
+            </div>
+          `,
+        );
+        const viewer = container.querySelector('nuxeo-image-viewer');
+        await viewerLoad(viewer);
+
+        sinon.spy(viewer._el, 'resize');
+        sinon.spy(viewer._el, 'zoomTo');
+
+        viewer.dispatchEvent(new CustomEvent('resize', { bubbles: true }));
+
+        expect(viewer._el.resize.notCalled).to.be.true;
+        expect(viewer._el.zoomTo.notCalled).to.be.true;
+      });
+
+      test("Should take resize and apply default zoom when viewer's parent is visible", async () => {
```

Comment:
Don't think the "take" adds anything here:
```suggestion
      test("Should resize and apply default zoom when viewer's parent is visible", async () => {
```

Reply:
It is kind of tricky, but yes we can assume that `not reset zoom` is enough to make clear that the image should not be reworked/changed.
Thanks!